pennylane/tape/qscript.py

```diff
@@ -834,25 +834,67 @@ def numeric_type(self) -> Union[type, tuple[type, ...]]:
     # Transforms: QuantumScript to QuantumScript
     # ========================================================
 
-    def copy(self, copy_operations: bool = False) -> "QuantumScript":
-        """Returns a shallow copy of the quantum script.
+    def copy(self, copy_operations: bool = False, **update) -> "QuantumScript":
+        """Returns a copy of the quantum script. If `update` was passed, the updated attributes are
+        modified, otherwise, all attributes match the original tape. The copy is a shallow copy if
+        `copy_operations` and `update` are both `False`.
 
         Args:
             copy_operations (bool): If True, the operations are also shallow copied.
                 Otherwise, if False, the copied operations will simply be references
                 to the original operations; changing the parameters of one script will likewise
-                change the parameters of all copies.
+                change the parameters of all copies. If any keyword arguments are passed to update,
+                this argument will be treated as True.
+
+        Keyword Args:
+            operations (Iterable[Operator]): An iterable of the operations to be performed. If provided, these
+                operations will replace the copied operations on the new tape.
+            measurements (Iterable[MeasurementProcess]): All the measurements to be performed. If provided, these
+                measurements will replace the copied measurements on the new tape.
+            shots (None, int, Sequence[int], ~.Shots): Number and/or batches of shots for execution. If provided, these
+                shots will replace the copied shots on the new tape.
+            trainable_params (None, Sequence[int]): the indices for which parameters are trainable. If provided, these
+                parameter indices will replace the copied parameter indicies on the new tape.
 
         Returns:
-            QuantumScript : a shallow copy of the quantum script
+            QuantumScript : a copy of the quantum script. If `update` was passed, the updated attributes are modified.
+
+        **Example**
```

Comment:
Can you add an example for how to use the keyword arguments?

Reply:
Isn't that already there? I used `measurements` specifically below, is there some element missing from that?

Reply:
Oh ooops I missed it.